code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm

```diff
@@ -618,3 +617,20 @@
 	var/obj/P = new chosen
 	if(P)
 		P.loc = get_turf(holder.my_atom)
+
+//Rainbow :o)
+/datum/chemical_reaction/slimeRNG
+	name = "Random Core"
+	id = "slimerng"
+	result = null
+	required_reagents = list("plasma" = 1)
+	result_amount = 1
+	required_other = 1
+	required_container = /obj/item/slime_extract/rainbow
+
+/datum/chemical_reaction/slimeRNG/on_reaction(var/datum/reagents/holder)
+	feedback_add_details("slime_cores_used","[replacetext(name," ","_")]")
+	var/mob/living/simple_animal/slime/S = new /mob/living/simple_animal/slime
+	S.colour = pick("grey","orange", "metal", "blue", "purple", "dark purple", "dark blue", "green", "silver", "yellow", "gold", "yellow", "red", "silver", "pink", "cerulean", "sepia", "bluespace", "pyrite", "light pink", "oil", "adamantine", "black")
```

Comment:
That should be fixed sometime
Op 27-jun.-2015 03:33 schreef "Incoming5643" <notifications@github.com>:

> In code/modules/reagents/Chemistry-Recipes/Slime_extracts.dm
> <https://github.com/tgstation/-tg-station/pull/10173#discussion_r33407875>
> :
>
> > @@ -618,3 +617,20 @@
> >  	var/obj/P = new chosen
> >  	if(P)
> >  		P.loc = get_turf(holder.my_atom)
> > +
> > +//Rainbow :o)
> > +/datum/chemical_reaction/slimeRNG
> > +	name = "Random Core"
> > +	id = "slimerng"
> > +	result = null
> > +	required_reagents = list("plasma" = 1)
> > +	result_amount = 1
> > +	required_other = 1
> > +	required_container = /obj/item/slime_extract/rainbow
> > +
> > +/datum/chemical_reaction/slimeRNG/on_reaction(var/datum/reagents/holder)
> > +	feedback_add_details("slime_cores_used","[replacetext(name," ","_")]")
> > +	var/mob/living/simple_animal/slime/S = new /mob/living/simple_animal/slime
> > +	S.colour = pick("grey","orange", "metal", "blue", "purple", "dark purple", "dark blue", "green", "silver", "yellow", "gold", "yellow", "red", "silver", "pink", "cerulean", "sepia", "bluespace", "pyrite", "light pink", "oil", "adamantine", "black")
>
> There's actually only the one type of slime, the colour var determines
> everything in terms of its name/apperance/drops so unfortunately you can't
> do a traditional typesof.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tgstation/-tg-station/pull/10173/files#r33407875>.
>